### DISCUSSION TITLE
RFC: Apps for Android, Android TV, iOS and Apple TV

### DISCUSSION THREAD

**Discussion by onedr0p** [midarrlabs/midarr-server#394] [Ideas]
I figured this would be a request at some point so might as well bring it up. I have no experience in app development so this can be used as a discussion if anyone wants to attempt such an epic.

I really love how Midarr works in that it offloads a lot of work to Sonarr and Radar so much that Midarr is pretty lightweight, e.g. no need for writing metadata services or having local storage.  I feel like having a app to watch movies and series on these platforms would bring in more adoption.

**Comment by trueChazza** [comment]
Thanks for opening this!

Interestingly I've recently looked into an Android TV app 😄

**Comment by trueChazza** [comment]
I think as a first step this would require fleshing out the API routes more and adding some documentation around it. This would allow anyone keen to tackle native apps, to have the resources they need.

**Comment by trueChazza** [comment]
I'm definitely keen to do an Android TV app!

**Comment by trueChazza** [comment]
https://developer.android.com/training/tv

Also from reading through Google's Android dev docs, Android TV and just normal Android apps share the same architecture - so doing one or the other should set us up nicely.

**Comment by trueChazza** [comment]
https://github.com/midarrlabs/kodi-client

We have a repo for a Kodi client with no progress so far - but as I mentioned, it's blocking on Midarr APIs being fleshed out abit more.

**Comment by trueChazza** [comment]
Also there is an alternative option too, where if we were to have the same sort of APIs as Jellyfin / Emby - Midarr could function on their native app offerings aswell.

But I'm not too keen on this. Short term it would get us up and running quickly - but long term it could possibly hinder or slow us down with future features.

**Comment by trueChazza** [comment]
https://github.com/orgs/midarrlabs/projects/1/views/1?filterQuery=api

There's a few issues in the backlog for progressing the APIs.

**Comment by bo0tzz** [comment]
For Immich, we've been using https://flutter.dev/ for our Android and iOS apps. Flutter can also target quite a few other platforms (including Android TV), so building a client in that could be a nice way of covering ground.

**Comment by bo0tzz** [comment]
And one other thought - this isn't anywhere near prod ready yet, but could be an option in the future if you want to stick close to Elixir: https://native.live/

**Reply by bo0tzz** [reply]
Here's a great talk from the recent ElixirConf about the current state of Liveview Native :) https://www.youtube.com/watch?v=xBi2Hj22qUU

**Reply by trueChazza** [reply]
Great talks at ElixirConf this year! I was watching this earlier today, very interesting. Thanks for linking 😁

**Comment by trueChazza** [comment]
#382 merged with api token support. This allows us to start moving forward with improved api routes etc - and then onto the Kodi client 🏄‍♂️

**Comment by trueChazza** [comment]
https://github.com/midarrlabs/midarr-server/discussions/389

Kodi add-on!

**Comment by trueChazza** [comment]
I'm going to move this into a discussion, as it's quite a broad topic. We can create issues tied to the discussion as we progress.

**Comment by hackmonker** [comment]
So typically we have a Kodi which is awesome mainly for tv and pc. For mobiles, not so much. So there here is another sugestion for mobilein case you were interested in the plugin/addon route. 

Cloudstream

https://cloudstream-on-fleek-co.ipns.dweb.link/
https://github.com/recloudstream/cloudstream/

This app works on plugins completely and felt like something of interest. Removes the need for a brand new native client. Just a suggestion

**Reply by trueChazza** [reply]
Awesome thank you! Sorry I just seen your comment now as I wasn’t notified for some reason. I’ll check this out
